src/tools/kata-ctl/src/utils.rs

```diff
@@ -38,6 +40,120 @@ pub fn get_kernel_version(proc_version_file: &str) -> Result<String> {
     Ok(kernel_version)
 }
 
+const OS_RELEASE: &str = "/etc/os-release";
+
+//Clear Linux has a different path (for stateless support)
+const OS_RELEASE_CLR: &str = "/usr/lib/os-release";
+
+const UNKNOWN: &str = "unknown";
+
+pub fn get_distro_details(os_release: &str, os_release_clr: &str) -> Result<(String, String)> {
+    let files = vec![os_release, os_release_clr];
+    let mut name = String::new();
+    let mut version = String::new();
+
+    for release_file in files.iter() {
+        match fs::read_to_string(release_file) {
+            Err(e) => {
+                if e.kind() == std::io::ErrorKind::NotFound {
+                    continue;
+                } else {
+                    return Err(anyhow!(
+                        "Error reading file {}: {}",
+                        release_file,
+                        e.to_string()
+                    ));
+                }
+            }
+            Ok(contents) => {
+                let lines = contents.lines();
+
+                let get_field = |line: &str| -> Result<String> {
+                    let fields: Vec<&str> = line.split("=").collect();
+                    if fields.len() < 2 {
+                        return Err(anyhow!("Unexpected file contents for {}", release_file));
+                    } else {
+                        let val = fields[1].trim();
+                        return Ok(String::from(val));
+                    }
+                };
+
+                for line in lines {
+                    if line.starts_with("NAME=") && name.is_empty() {
```

Comment:
Should `NAME` and `VERSION_ID` be defined as const like `ARCH_CPU_VENDOR_FIELD`?

Reply:
These are not architecture specific, hence I havent defined them as other architecture specific fields.